File: frontend/src/api/knowledge.ts
import request from './request'

export interface Document {
  id: number
  filename: string
  file_type: string
  file_size: number | null
  chunk_count: number
  uploaded_at: string
}

export interface SearchResult {
  content: string
  score: number
  file_id: number | null
  filename: string | null
  chunk_index: number | null
}

export interface KnowledgeSearchResponse {
  query: string
  results: SearchResult[]
}

export function uploadDocument(file: File) {
  const form = new FormData()
  form.append('file', file)
  return request.post<Document>('/api/knowledge/upload', form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export function getDocuments() {
  return request.get<Document[]>('/api/knowledge/documents')
}

export function deleteDocument(id: number) {
  return request.delete(`/api/knowledge/documents/${id}`)
}

export function searchKnowledge(query: string, k = 5) {
  return request.get<KnowledgeSearchResponse>('/api/knowledge/search', {
    params: { query, k },
  })
}

File: backend/app/schemas/knowledge.py
"""知识库相关 Pydantic 模型"""

from datetime import datetime
from pydantic import BaseModel


class DocumentOut(BaseModel):
    id: int
    filename: str
    file_type: str
    file_size: int | None = None
    chunk_count: int = 0
    uploaded_at: datetime

    class Config:
        from_attributes = True


class SearchResult(BaseModel):
    content: str
    score: float
    file_id: int | None = None
    filename: str | None = None
    chunk_index: int | None = None


class KnowledgeSearchOut(BaseModel):
    query: str
    results: list[SearchResult]


class KnowledgeAskOut(BaseModel):
    query: str
    answer: str
    sources: list[SearchResult]
